cmd/dataobj-inspect/main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/dustin/go-humanize"
+	"github.com/grafana/loki/v3/pkg/dataobj"
+)
+
+func main() {
+	flag.Parse()
+
+	for _, f := range flag.Args() {
+		printFile(f)
+	}
+}
+
+func printFile(filename string) {
+	f, err := os.Open(filename)
+	if err != nil {
+		log.Printf("%s: %v", filename, err)
+		return
+	}
+	defer func() { _ = f.Close() }()
+
+	objBytes, err := io.ReadAll(f)
+	if err != nil {
+		log.Printf("failed to read file: %v", err)
+		return
+	}
+
+	reader := dataobj.Decoder(bytes.NewReader(objBytes))
+
+	sections, err := reader.Sections(context.Background())
+	if err != nil {
+		log.Printf("failed to read sections: %v", err)
+		return
+	}
+
+	for _, section := range sections {
+		switch int(section.Type) {
+		case 2:
```

Comment:
this whole switch is pretty ugly.
All the metadata types are internal to the dataobj package, so I can't actually import them.

I briefly attempted to make an inspector local to the package have functions for custom behaviour, but I couldn't pass `logsmd.ColumnInfo` as a parameter to a the custom func as they are also internal. `func ForEachLogsColumn(f func(col logsmd.ColumnInfo))` doesn't work.

I think the options we have without exposing the internals is build an "inspection" object which confirms to a new public interface or to hard-code the actual print statements for debugging.

Suggestions welcome!